be/src/olap/rowset/segment_v2/column_reader.h

```diff
@@ -103,6 +106,9 @@ class ColumnReader {
 
     // column zone map info
     std::unique_ptr<ColumnZoneMap> _column_zone_map;
+
+    // keep dict page
+    std::shared_ptr<BinaryPlainPageDecoder> _column_dict_page_decoder;
```

Comment:
why shared_ptr? I think unique_ptr is enough.

Reply:
one dict page may correspond to many data pages,so _column_dict_page_decoder ptr may be share by many BinaryDictPageDecoder

Reply:
ColumnReader is owned by Segment and Rowset, being alive even when there is no query. Therefore it should retain memory as small as possible. I think it'd better to keep dict page decoder in `FileColumnIterator`.

Reply:
👌